nemo/utils/exp_manager.py

```diff
@@ -281,17 +287,25 @@ def exp_manager(trainer: 'pytorch_lightning.Trainer', cfg: Optional[Union[DictCo
     logging.info(f'Experiments will be logged at {log_dir}')
     trainer._default_root_dir = log_dir
 
+    if cfg.log_local_rank_0_only and cfg.log_global_rank_0_only:
+        raise ValueError(
+            f"Cannot set both log_local_rank_0_only and log_global_rank_0_only to True. Please set either one or neither."
+        )
+
+    # This is set if the env var NEMO_TESTING is set to True.
+    nemo_testing = get_envbool(NEMO_ENV_VARNAME_TESTING, False)
+
     # Handle logging to file
-    if get_envbool(NEMO_ENV_VARNAME_TESTING, False) or world_size <= 32:
-        # If NEMO_TESTING is set (debug mode) or if less than 32 ranks save all log files
+    # Logs local rank 0 only
+    if local_rank == 0 and cfg.log_local_rank_0_only and not nemo_testing:
```

Comment:
Dont use implicit truths like this. Yes we know the flag is true or false but don't check it like this. Just use cfg.log_local_... is True/False as you need it here.